ludwig/predict.py

```diff
@@ -107,7 +101,22 @@ def full_predict(
     model.close_session()
 
     if is_on_master():
-        os.mkdir(experiment_dir_name)
+        # setup directories and file names
+        experiment_dir_name = output_directory
+        suffix = 0
+        while os.path.exists(experiment_dir_name):
+            experiment_dir_name = output_directory + '_' + str(suffix)
+            suffix += 1
```

Comment:
We have this logic repeated in a few places. Should we write a util method?

Reply:
good idea

Reply:
done